### PR TITLE
Adding pyflakes and black to the CI

### PR DESCRIPTION
**Pull request by MrRobot2211**
I will try out adding pyflakes and back for style-checking and formatting.

### CODE REVIEW

**Comment by MrRobot2211** [comment]
Onw alternative would be to do something similar to this

https://github.com/unitaryfund/mitiq/blob/master/.github/workflows/test.yml

File: .github/workflows/tests-gpu.yml [hunk 1]
```diff
@@ -83,6 +84,18 @@ jobs:
...
         run: coveralls --service=github
 
+  pyflakes:
+    name: runner/pyflakes
+    needs: cases
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: pyflakes
+        uses: reviewdog/action-pyflakes@master
```
Comment (hodgestar) [inline .github/workflows/tests-gpu.yml:94]: I would prefer not to use a complex third party tool like reviewdog and to just run pyflakes directly ourselves.

Reviewdog (and similar tools like CodeClimate) add complexity for little benefit and some significant downsides (e.g. it becomes much harder to answer a simple question like "which version of pyflakes with what configuration is expected to work".

Comment (MrRobot2211) [reply .github/workflows/tests-gpu.yml:94]: Good point what do you think of using an scheme like the one on mitiq https://github.com/unitaryfund/mitiq/blob/master/.github/workflows/test.yml ? Maybe the calls to make overcomplexify it a ittle bit an wer can directly make the calls on the yml

Comment (hodgestar) [reply .github/workflows/tests-gpu.yml:94]: Let's try put the calls to pyflakes and black directly into the .yml to start with. If things get complicated, we can think about adding something like `make`.

**Comment by MrRobot2211** [comment]
Well it breaks as expected :).
@hodgestar do you have any opinions on which erros should be purposefully skip?

I am tempted to just borrow this https://github.com/unitaryfund/mitiq/blob/master/.flake8 ,
but I prefer to hear your opinion.

**Comment by hodgestar** [comment]
@MrRobot2211 I applied my own personal preferences (i.e. no skipping any flake8 errors for now but accepting black's suggested line length of 88 characters) and pushed the necessary flake8 and black changes to (hopefully) have the checks pass successfully.

Let me know if you like it or not.

We can also try add `isort` as a linting tool. That takes care of ordering imports, which is quite nice, but can be a little finicky (e.g. it needs to know which imports are from "third-party" sources -- i.e. from outside the current package) so there is sometimes a bit of fiddling needed.

**Review by hodgestar — APPROVED** [review]
Approved, but @MrRobot2211 and the GitHub Actions should be happy to before we merge.

**Comment by MrRobot2211** [comment]
Thank you @hodgestar. This looks considerably better. Agreed on leaving `isort` for a sometime in the future PR.

**Pull request merged**